### DISCUSSION TITLE
feat defineDirective

### DISCUSSION THREAD

**Discussion by liuseen-l** [vuejs/rfcs#653] [RFC Discussions]
Although vue provides `Directive` for users to get type hints when defining custom directives, it requires users to manually import them. Therefore, adding this micro can avoid this operation. Secondly, when the amount of code becomes large, vue instructions are mixed with other variables and it is difficult to distinguish them, which can be well distinguished by micro definitions

Look forward to your different perspectives

Pr address: https://github.com/vuejs/core/pull/10512/files

**Comment by posva** [comment]
Replacing a perfectly correct and simple type annotation with a function wrap doesn’t sound like a good idea because it moves away from the already existing and semantic way of doing things without bringing any objective value
